controllers/volsync/vshandler.go

```diff
@@ -767,6 +767,28 @@ func (v *VSHandler) DeleteRD(pvcName string) error {
 	return nil
 }
 
+func (v *VSHandler) DeleteSnapshots(pvcNamespace string) error {
+	// Remove a Snapshot by name that is owned (by parent vrg owner)
+	snapList := &snapv1.VolumeSnapshotList{}
+
+	err := v.listByOwner(snapList, pvcNamespace)
+	if err != nil {
+		return err
+	}
+
+	for i := range snapList.Items {
+		snapshot := snapList.Items[i]
+		// Delete the VolumeSnapshot, log errors with cleanup but continue on
+		if err := v.client.Delete(v.ctx, &snapshot); err != nil {
+			v.log.Error(err, "Error cleaning up VolumeSnapshot", "name", snapshot.GetName())
+		} else {
+			v.log.Info("Deleted VolumeSnapshot", "name", snapshot.GetName())
+		}
```

Comment:
This works, but it would be more idomatic to handle the error as special case, and the normal flow as normal case:

    if err != nil {
        log error...
        continue
    }

    Normal flow here...